### DISCUSSION TITLE
includes doesn't work with $(env VARIABLE)

### DISCUSSION THREAD

**Discussion by mactul** [xmake-io/xmake#5266] [Q&A]
Hello !

I have a set of functions in the description scope that I use in all my xmake.lua

I want to write something like this

```lua
includes("$(env XMAKE_TOOLS)/tool_xxx.lua")
```
But unfortunately, it seems that `includes` doesn't support the `$(env XXX)` syntax.
I can't write the absolute path because the project is meant to be built on multiple machines and multiple operating systems, so the absolute path is always different...

Does a workaround exists to solve my problem or am I screwed ?

Thank you in advance,
Mactul

**Comment by waruqi** [comment]
```lua
includes(path.join(os.getenv(XMAKE_TOOLS), "tool_xxx.lua"))
```
